src/utils/StorageHelper.ts

```diff
@@ -0,0 +1,83 @@
+import { getPublic, sign } from "@toruslabs/eccrypto";
+import { get, post } from "@toruslabs/http-helpers";
+import { decryptData, encryptData, keccak256 } from "@toruslabs/metadata-helpers";
+
+import { LoginDetails } from "../handlers/interfaces";
+import { REDIRECT_PARAMS_STORAGE_METHOD, REDIRECT_PARAMS_STORAGE_METHOD_TYPE } from "./enums";
+import { are3PCSupported, storageAvailable } from "./helpers";
+
+class StorageHelper {
+  private currentStorageMethod: REDIRECT_PARAMS_STORAGE_METHOD_TYPE = REDIRECT_PARAMS_STORAGE_METHOD.LOCAL_STORAGE;
+
+  private isInitialized = false;
+
+  private storageServerUrl = "https://broadcast-server.tor.us";
+
+  constructor(serverUrl: string) {
+    this.storageServerUrl = serverUrl;
+  }
+
+  init() {
+    const support = are3PCSupported();
+    const localStorageAvailable = storageAvailable(REDIRECT_PARAMS_STORAGE_METHOD.LOCAL_STORAGE);
+    if (support && localStorageAvailable) {
+      // use local storage as default for storing stuff
+      this.currentStorageMethod = REDIRECT_PARAMS_STORAGE_METHOD.LOCAL_STORAGE;
+    } else {
+      // use server store as default for storing stuff
+      this.currentStorageMethod = REDIRECT_PARAMS_STORAGE_METHOD.SERVER;
+    }
+    this.isInitialized = true;
+  }
+
+  async storeLoginDetails(params: LoginDetails, scope: string): Promise<void> {
```

Comment:
the nonce that we are using as scope here is a randomId is marked to not use for cryptographic purposes. So we should avoid using that.

https://github.com/torusresearch/CustomAuth/blob/b1f7062a8bbcc9961ad0839570346a2e0f1d8d83/src/utils/helpers.ts#L45

Reply:
That's correct. Hence, we hash it and use it since we need a deterministic key before & after redirect

Reply:
got it